docker-native packaging not working with Java 17 and maven

### Expected Behavior

GIven a brand new project generated from CLI or Launch with JDK_17 as the javaVersion
When you execute a `./mvnw clean package -Dpackaging=docker-native` build
It should complete successfully

### Actual Behaviour

```
[ERROR] Failed to execute goal io.micronaut.build:micronaut-maven-plugin:3.0.2:docker-native (default-docker-native) on project test: To build native images you must set the Java target byte code level to Java 11 or below
```



### Steps To Reproduce

```
curl --location --request GET 'https://snapshot.micronaut.io/create/default/com.test?lang=JAVA&build=MAVEN&test=SPOCK&javaVersion=JDK_17' --output test.zip

unzip test.zip

cd test

./mvnw clean package -Dpackaging=docker-native
```


This error stems from this code:  [DockerNativeMojo.java#L97](https://github.com/micronaut-projects/micronaut-maven-plugin/blob/97191b483230223990b5d03c7e806eaf5c0fd201/src/main/java/io/micronaut/build/DockerNativeMojo.java#L97)

Removing this ^^ code and updating the `from` image to something 17 compatible (eg, `ghcr.io/graalvm/graalvm-ce:java17-21.3`) should get this working.  This still defaults to `java11` here: [AbstractDockerMojo.java#L30](https://github.com/micronaut-projects/micronaut-maven-plugin/blob/97191b483230223990b5d03c7e806eaf5c0fd201/src/main/java/io/micronaut/build/AbstractDockerMojo.java#L30)

### Environment Information

MacOS
JDK 17

### Example Application

_No response_

### Version

3.1.1 and 3.1.2-SNAPSHOT

--- a/src/main/java/io/micronaut/build/AbstractDockerMojo.java
+++ b/src/main/java/io/micronaut/build/AbstractDockerMojo.java
@@ -27,11 +27,11 @@ import static io.micronaut.build.services.ApplicationConfigurationService.DEFAUL
  * Abstract base class for mojos related to Docker files and builds.
  *
  * @author Álvaro Sánchez-Mariscal
+ * @author Iván López
  * @since 1.1
  */
 public abstract class AbstractDockerMojo extends AbstractMojo {
 
-    public static final String DEFAULT_GRAAL_DOCKER_VERSION = "java11";
     public static final String LATEST_TAG = "latest";
     public static final String DEFAULT_BASE_IMAGE_GRAALVM_RUN = "frolvlad/alpine-glibc:alpine-3.12";
     public static final String MOSTLY_STATIC_NATIVE_IMAGE_GRAALVM_FLAG = "-H:+StaticExecutableWithDynamicLibC";
@@ -111,9 +111,9 @@ public abstract class AbstractDockerMojo extends AbstractMojo {
     protected String getFrom() {
         if (staticNativeImage) {
             // For building a static native image we need a base image with tools (cc, make,...) already installed
-            return jibConfigurationService.getFromImage().orElse("ghcr.io/graalvm/graalvm-ce:" + DEFAULT_GRAAL_DOCKER_VERSION + "-" + graalVmVersion());
+            return jibConfigurationService.getFromImage().orElse("ghcr.io/graalvm/graalvm-ce:" + graalVmJvmVersion() + "-" + graalVmVersion());
         } else {
-            return jibConfigurationService.getFromImage().orElse("ghcr.io/graalvm/native-image:" + DEFAULT_GRAAL_DOCKER_VERSION + "-" + graalVmVersion());
+            return jibConfigurationService.getFromImage().orElse("ghcr.io/graalvm/native-image:" + graalVmJvmVersion() + "-" + graalVmVersion());
         }
     }
 

--- a/src/main/java/io/micronaut/build/DockerNativeMojo.java
+++ b/src/main/java/io/micronaut/build/DockerNativeMojo.java
@@ -28,6 +28,7 @@ import java.util.Set;
  * <pre>mvn package -Dpackaging=docker-native</pre>
  *
  * @author Álvaro Sánchez-Mariscal
+ * @author Iván López
  * @since 1.1
  */
 @Mojo(name = DockerNativeMojo.DOCKER_NATIVE_PACKAGING, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
@@ -93,8 +94,8 @@ public class DockerNativeMojo extends AbstractDockerMojo {
     }
 
     private void checkJavaVersion() throws MojoExecutionException {
-        if (javaVersion().getMajorVersion() > 11) {
-            throw new MojoExecutionException("To build native images you must set the Java target byte code level to Java 11 or below");
+        if (javaVersion().getMajorVersion() > 17) {
+            throw new MojoExecutionException("To build native images you must set the Java target byte code level to Java 17 or below");
         }
     }
 

